rust/lance-index/Cargo.toml

```
@@ -50,6 +50,7 @@ serde_json.workspace = true
 serde.workspace = true
 snafu.workspace = true
 tantivy.workspace = true
+tantivy-jieba.workspace = true
```

Could we make this an optional dependency for `lance-index`?

---

@wjones127 Is there any example about optional dependency for other rust lib? I'm not quite familiar with how to achieve it.

---

https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies

```suggestion
tantivy-jieba = { workspace = true, optional = true }
```